### ISSUE TITLE
Bug with Range Picker Component (Nebular v5.0.0)

### ISSUE THREAD

**Issue by jlopez994** [akveo/nebular#2303]
### Issue type

**I'm submitting a ...**  (check one with "x")

* [X] bug report
* [ ] feature request

### Issue description

**Current behavior:**

If you select a range and then you try to select another range from another month, you get this error: `t.getFullYear is not a function` and months of the year are not shown.

**Expected behavior:**

Months of the year should be shown when I try to select another range.

**Steps to reproduce:**

<img src="https://s5.gifyu.com/images/InstantGif_2020.03.30-17.51.gif" />

### Other information:

Tested with Google Chrome v80.0.3987.149 and Nebular v5.0.0

**Comment by bkeresztes** [comment]
Hi,

I have the same issue. Not sure if it helps, but the issue is here:
NbNativeDateService.prototype.isSameYear = function (date1, date2) {
        return date1.getFullYear() === date2.getFullYear();
    };
for some reason the date2 is the range itself not the 'end' field of the range.

**Comment by bkeresztes** [comment]
I have managed to find a workaround while this gets fixed. You can put this in your AppModule constructor to overwrite the wrong part:

    Object.defineProperty(NbCalendarMonthCellComponent.prototype, 'selected', {
      get: function () {
          return this.dateService.isSameMonthSafe(this.date, this.selectedValue.end);
      },
      enumerable: true,
      configurable: true
    });

In the original the this.selectedValue is passed in the second parameter which is the range. This works for me, hope it helps.

**Comment by jlopez994** [comment]
> I have managed to find a workaround while this gets fixed. You can put this in your AppModule constructor to overwrite the wrong part:
> 
> ```
> Object.defineProperty(NbCalendarMonthCellComponent.prototype, 'selected', {
>   get: function () {
>       return this.dateService.isSameMonthSafe(this.date, this.selectedValue.end);
>   },
>   enumerable: true,
>   configurable: true
> });
> ```
> 
> In the original the this.selectedValue is passed in the second parameter which is the range. This works for me, hope it helps.

Thanks! It worked partially. There is still another bug. Sometimes I get this one:
` Cannot read property 'end' of undefined`
when I try to select a month from another year.

**Comment by jlopez994** [comment]
> > I have managed to find a workaround while this gets fixed. You can put this in your AppModule constructor to overwrite the wrong part:
> > ```
> > Object.defineProperty(NbCalendarMonthCellComponent.prototype, 'selected', {
> >   get: function () {
> >       return this.dateService.isSameMonthSafe(this.date, this.selectedValue.end);
> >   },
> >   enumerable: true,
> >   configurable: true
> > });
> > ```
> > 
> > 
> > In the original the this.selectedValue is passed in the second parameter which is the range. This works for me, hope it helps.
> 
> Thanks! It worked partially. There is still another bug. Sometimes I get this one:
> ` Cannot read property 'end' of undefined`
> when I try to select a month from another year if I haven't selected any range before.

It can be fixed adding this verification:

`this.selectedValue ? this.selectedValue.end : new Date()`

So the result would be
```
Object.defineProperty(NbCalendarMonthCellComponent.prototype, 'selected', {
  get() {
    return this.dateService.isSameMonthSafe(
      this.date,
      this.selectedValue ? this.selectedValue.end : new Date()
    );
  },
  enumerable: true,
  configurable: true
});
```

**Comment by yggg** [comment]
The issue was fixed by #2330 (part of the `6.1.0` release). The problem was that `NbCalendarRangeComponent` was using `NbCalendarMonthCellComponent` as cell component instead of `NbCalendarRangeDayCellComponent`. After the update to the fixed version, you can remove lines with patching `NbCalendarMonthCellComponent.prototype` as it would break regular picker.
